lib/ansible/module_utils/facts/timeout.py

```diff
@@ -25,7 +25,11 @@
 DEFAULT_GATHER_TIMEOUT = 10
 
 
-class TimeoutError(Exception):
+class AnsibleTimeoutError(Exception):
+    pass
+
+class TimeoutError(AnsibleTimeoutError):
+    # TODO: figure out how to issue deprecation warning here.
     pass
```

Comment:
This is the wrong thing to do.  Instead,
``` python
try:
    class TimeoutError(TimeoutError):
        pass
except NameError:
    # Compatibility with Python 2 which doesn't have a TimeoutError of its own
    class TimeoutError(Exception):
        pass
```
We then raise this TimeoutError but calling code can use the python3 version of TimeoutError if their code only runs on Python3.

https://youtu.be/o9pEzgHorH0?t=615

Reply:
no, i specifically don't want to use an exception anything else throws

Reply:
Well, I can't help you when you want to do things the wrong way.

Reply:
Also, people will still be able to catch the exceptions separately if they want to because exception catching is basically a switch statement.

Reply:
No, you are incorrect.  The exceptions are distinct no matter what and they can choose to catch either or only one.  Apart from the style problems, this PR seems to be based on a false understanding of Python and should just be closed.